api/v1alpha1/hostedclusterinfrastructure_types.go

```diff
@@ -0,0 +1,344 @@
+
+	// Platform specifies the underlying infrastructure provider for the cluster
+	// and is used to configure platform specific behavior.
+	//
+	// +immutable
+	Platform PlatformInfraSpec `json:"platform"`
+
+	// baseDomain is the base domain of the cluster. All managed DNS records will
+	// be sub-domains of this base.
+	//
+	// For example, given the base domain `openshift.example.com`, an API server
+	// DNS record may be created for `cluster-api.openshift.example.com`.
+	//
+	// Once set, this field cannot be changed.
+	BaseDomain string `json:"baseDomain"`
+
+	// CloudProvider secret, contains the Cloud credenetial and Base Domain
+	// When not present, we expect all values to populated at create time
+	// This can be from the hypershift cli or via a kubectl create.
+	// +optional
+	CloudProvider corev1.LocalObjectReference `json:"cloudProvider,omitempty"`
+}
+
+// PlatformSpec specifies the underlying infrastructure provider for the cluster
+// and is used to configure platform specific behavior.
+type PlatformInfraSpec struct {
+	// Type is the type of infrastructure provider for the cluster.
+	//
+	// +kubebuilder:validation:Enum=AWS;Azure;PowerVS
+	// +immutable
+	Type PlatformType `json:"type"`
+
+	// AWS specifies configuration for clusters running on Amazon Web Services.
+	//
+	// +optional
+	// +immutable
+	AWS *AWSPlatformInfraSpec `json:"aws,omitempty"`
+
+	// Azure defines azure specific settings
+	Azure *AzurePlatformSpec `json:"azure,omitempty"`
+
+	// PowerVS specifies configuration for clusters running on IBMCloud Power VS Service.
+	// This field is immutable. Once set, It can't be changed.
+	//
+	// +optional
+	// +immutable
+	PowerVS *PowerVSPlatformSpec `json:"powervs,omitempty"`
+}
+
+// AWSPlatformSpec specifies configuration for clusters running on Amazon Web Services.
+type AWSPlatformInfraSpec struct {
+	// Region is the AWS region in which the cluster resides. This configures the
+	// OCP control plane cloud integrations, and is used by NodePool to resolve
+	// the correct boot AMI foHostedClusterInfrastructureConfigStatus a given release.
+	// HostedCluster.spec.platform.aws.region
+	//
+	// +immutable
+	Region string `json:"region"`
+
+	// ResourceTags is a list of additional tags to apply to AWS resources created
+	// for the cluster. See
+	// https://docs.aws.amazon.com/general/latest/gr/aws_tagging.html for
+	// information on tagging AWS resources. AWS supports a maximum of 50 tags per
+	// resource. OpenShift reserves 25 tags for its use, leaving 25 tags available
+	// for the user.
+	//
+	// +kubebuilder:validation:MaxItems=25
+	// +optional
+	ResourceTags []AWSResourceTag `json:"resourceTags,omitempty"`
+
+	// Zones are availability zones in an AWS region.
+	//
+	// +optional
+	Zones []string `json:"zones,omitempty"`
+}
+
+// HostedClusterInfrastructureConfigStatus reports the IDs of the platform resources via a status field
+type HostedClusterInfrastructureConfigStatus struct {
```

Comment:
Shouldn't this be HostedClusterInfrastructureStatus

Reply:
`HostedClusterInfrastructureStatus` is already used for the base status struct around Ln 240